### ISSUE TITLE
Bulk inserts are noticably slower on PyPy

### ISSUE THREAD

**Issue by thedrow** [chtd/psycopg2cffi#44]
See the logs from a [PyPy build](https://travis-ci.org/chtd/psycopg2cffi/jobs/67628694#L806) and the logs from a [CPython 3.4 build](https://travis-ci.org/chtd/psycopg2cffi/jobs/67628693#L847).
It's pretty consistent across builds and also reproduces itself in all CPython versions.

**Comment by lopuhin** [comment]
This is very interesting, thank you. I will check locally a bit later today, and see what can be improved.

**Comment by thedrow** [comment]
I suspect that this is because you're not warming up the JIT.
You should run at least 10,000 (depending on the operation. It could be more or could be less) loops before starting to measure the performance of a feature.

**Comment by lopuhin** [comment]
Ah, you are right, and that is why there are several runs repeated - it is much faster below https://travis-ci.org/chtd/psycopg2cffi/jobs/67628694#L832 - so I will only report times after warmup, what do you think?

**Comment by thedrow** [comment]
Yes but you should run those benchmarks the same number of time as you do for CPython.
So warm up and then run the benchmarks 5 times.
